### PR TITLE
Trying to fix #2112. 

### LINKED ISSUE #2112: unkool warning
Trying to dig more into the warnings reported in #2109, I finally identified this nasty little bugger:

```
module TEST
  syntax Type ::= Id [klabel("class")]
  configuration <k> .K </k>
endmodule
```

Kompiling it, you get this:

```
[Warning] Compiler: Could not find main syntax module with name TEST-SYNTAX in
definition.  Use --syntax-module to specify one. Using TEST as default.
[Warning] Inner Parser: Parsing ambiguity. Arbitrarily choosing the first.
1: syntax KBott ::= "."
[Location(#token("48",Int),#token("20",Int),#token("48",Int),#token("80",Int))
Source(#token("D:\\github\\grosu\\k\\k-distribution\\target\\release\\k\\lib\\java\\..\\..\\include\\builtin\\kast.k",KString))
hook(#token("org.kframework.kore.EmptyK",AttributeValue))
klabel(#token("#EmptyK",AttributeValue)) klabel(#token("#EmptyK",KString))]
    #EmptyK()
2: syntax Type ::= Id
[Location(#token("2",Int),#token("19",Int),#token("2",Int),#token("38",Int))
Source(#token("D:\\github\\grosu\\k\\k-distribution\\tutorial\\2_languages\\2_kool\\2_typed\\2_static\\.\\test.k",KString))
klabel(#token("class",AttributeValue)) klabel(#token("class",KString))]
    class(#EmptyK())
        Source(D:\github\grosu\k\k-distribution\tutorial\2_languages\2_kool\2_typed\2_static\.\test.k)
        Location(3,21,3,22)
```

As a user, here I throw the towel.  :)

### PR DESCRIPTION
**Pull request by radumereuta**
Removing chain ambiguities that are not allowed to appear under automatically added subsorts.

Testing, do not review.

### CODE REVIEW

**Comment by radumereuta** [comment]
Jenkins: test this please.

**Comment by radumereuta** [comment]
Jenkins: test this please.

**Comment by cos** [comment]
Jenkins: test this please

**Comment by radumereuta** [comment]
Finally managed to fix all tests. I was able to successfully run all the tests.
I worked with @cos on the fix, so @andreistefanescu please review.
@grosu I had to modify kool to make it work, so please have a look.

File: k-distribution/tutorial/2_languages/2_kool/2_typed/2_static/kool-typed-static.k [hunk 1]
```diff
@@ -110,8 +110,7 @@ the wrapper in the generated documentation, we associate it an
 //@ \subsection{Types}
 
   syntax Type ::= "void" | "int" | "bool" | "string"
-                | Id      // TODO: switch to next once #2112 fixed
-//                | Id                     [klabel("class")]  // see next
+                | Id                     [klabel("class"), avoid]  // see next
```
Comment (grosu) [inline k-distribution/tutorial/2_languages/2_kool/2_typed/2_static/kool-typed-static.k:113]: This seems to be the only change in KOOL, which definitely LGTM.

**Comment by andreistefanescu** [comment]
lgtm

**Pull request merged**